File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        
        Map<String, Integer> closet = new HashMap<>();
        for(String[] info: clothes) {
            String type = info[1];
            closet.put(type, closet.getOrDefault(type,0)+1);
        }
        
        for(int count: closet.values()) {
            answer *= (count + 1);
        }
        return answer-1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.swift
import Foundation

func solution(_ clothes:[[String]]) -> Int {
    var dict: [String:Int] = [:]
    for clothe in clothes {
        let type = clothe[1]
        dict[type, default:0] += 1
    }
    var answer = 1
    for val in dict.values {
        answer *= (val+1)
    }
    return answer-1
}